tests/unit/package_managers/test_gomod.py

```diff
@@ -188,33 +203,52 @@ def test_resolve_gomod(
 
     run_side_effects.append(
         proc_mock(
-            "go list -e -mod readonly -m",
+            "go list -e -m",
             returncode=0,
             stdout=get_mocked_data(data_dir, f"{mocked_data_folder}/go_list_modules.json").replace(
                 "{repo_dir}", str(module_dir)
             ),
         )
     )
-    run_side_effects.append(
-        proc_mock(
-            "go list -e -mod readonly -deps -json all",
-            returncode=0,
-            stdout=get_mocked_data(data_dir, f"{mocked_data_folder}/go_list_deps_all.json"),
-        )
-    )
-    run_side_effects.append(
-        proc_mock(
-            "go list -e -mod readonly -deps -json ./...",
-            returncode=0,
-            stdout=get_mocked_data(data_dir, f"{mocked_data_folder}/go_list_deps_threedot.json"),
-        )
-    )
     mock_run.side_effect = run_side_effects
+    mock_go_list_deps.side_effect = [
+        _parse_go_list_deps_data(data_dir, f"{mocked_data_folder}/go_list_deps_all.json"),
+        _parse_go_list_deps_data(data_dir, f"{mocked_data_folder}/go_list_deps_threedot.json"),
+    ]
 
     mock_version_resolver.get_golang_version.return_value = "v0.1.0"
     mock_go_release.return_value = "go0.1.0"
     mock_get_gomod_version.return_value = ("0.1.1", "0.1.2")
 
+    parse_packages_mocked_data: list[ParsedPackage] = []
+    mock_parse_packages.return_value = parse_packages_mocked_data
+
+    if has_workspaces:
```

Comment:
This makes an already complex test even more complex. What do you think about moving it to a separate test-case? Also judging from the amount of mocking that is necessary this should probably be better tested with an integration test.

Reply:
> This makes an already complex test even more complex. What do you think about moving it to a separate test-case?

This will prolong the feature delivery even more IMO. That test is a beast. I already have another branch with more refactoring that wasn't directly connected to this work making this function rely even more on mocking and instead test the individual pieces of logic in isolation. That is not possible though yet, so my plan was:

- migrate everything this test does to invidividual isolated tests
- get rid of the `test_resolve_gomod` test to leave the rest to the integration test suite as you correctly point out

We're just not ^there yet, so if possible, I'd like to keep it this way for the time being with the promise that I'll get rid of the test eventually when I'm done with the module.

Reply:
Fair enough, splitting this one is an effort of its own, I won't block on it.